### PR TITLE
Replace Set with Collection in SoundRegistry

### LINKED ISSUE #38: Replace Set with Collection in SoundRegistry
This method requires Set
https://github.com/unnamed/creative/blob/1bc8834493f9b8dac5db1b5c19ccdbe8d50eaa12/api/src/main/java/team/unnamed/creative/sound/SoundRegistry.java#L57

but this constructor can accept any collection
https://github.com/unnamed/creative/blob/1bc8834493f9b8dac5db1b5c19ccdbe8d50eaa12/api/src/main/java/team/unnamed/creative/sound/SoundRegistryImpl.java#L54

because of this I have to convert my collection to a Set which is a waste

### PR DESCRIPTION
**Pull request by MelonHell**
resolves #38

### CODE REVIEW

**Comment by yusshu** [comment]
Instead of changing the old methods, this should add new methods *(don't mark the old ones as deprecated)*, so backwards-compatibility is kept. Also note that `@since` tags for these new methods should be set to `1.6.0`

**Comment by MelonHell** [comment]
Replacing Set with Collection will not break backward compatibility since Set inherits Collection

**Comment by yusshu** [comment]
It breaks already compiled code, as it changes the method signature

File: api/src/main/java/team/unnamed/creative/sound/SoundRegistry.java [hunk 1]
```diff
@@ -84,6 +97,22 @@ public interface SoundRegistry extends ResourcePackPart, Namespaced, Examinable
...
+     * @return A new sound registry instance
+     * @since 1.6.0
+     * @deprecated Use {@link #soundRegistry(String, Collection)} instead
+     */
+    @Deprecated
+    @ApiStatus.ScheduledForRemoval(inVersion = "2.0.0")
+    static @NotNull SoundRegistry of(final @NotNull String namespace, final @NotNull Collection<SoundEvent> sounds) {
+        return soundRegistry(namespace, sounds);
+    }
+
```
Comment (yusshu) [inline api/src/main/java/team/unnamed/creative/sound/SoundRegistry.java:115]: Remove this, we don't want to add deprecated methods

Comment (MelonHell) [reply api/src/main/java/team/unnamed/creative/sound/SoundRegistry.java]: oops, I didn't notice it was Deprecated

**Pull request merged**